stable/spinnaker/templates/statefulsets/halyard.yaml

```diff
@@ -0,0 +1,110 @@
+spec:
+  serviceName: {{ template "spinnaker.fullname" . }}-halyard
+  replicas: 1
+  selector:
+    matchLabels:
+{{ include "spinnaker.standard-selector-labels" . | indent 6 }}
+      component: halyard
+  template:
+    metadata:
+      labels:
+{{ include "spinnaker.standard-labels" . | indent 8 }}
+        component: halyard
+    spec:
+      serviceAccountName: {{ template "spinnaker.fullname" . }}-halyard
+      securityContext:
+        runAsUser: 1000
+        fsGroup: 1000
+      initContainers:
+      - name: "create-halyard-local"
+        image: {{ .Values.halyard.image.repository }}:{{ .Values.halyard.image.tag }}
+        command:
+        - bash
+        - -c
+        - |
+          # Override Halyard daemon's listen address
+          cp /opt/halyard/config/* /tmp/config && \
+          printf 'server.address: 0.0.0.0\n' > /tmp/config/halyard-local.yml && \
+          # Use Redis deployed via the dependent Helm chart
+          mkdir -p /tmp/spinnaker/.hal/default/service-settings && \
+          printf 'overrideBaseUrl: redis://:{{ .Values.redis.redisPassword }}@{{ .Release.Name }}-redis:6379\nskipLifeCycleManagement: true\n' > /tmp/spinnaker/.hal/default/service-settings/redis.yml && \
+          # Route the /gate path of Deck to Gate
+          printf 'env:\n  API_HOST: http://spin-gate.{{ .Release.Namespace }}:8084/\n' > /tmp/spinnaker/.hal/default/service-settings/deck.yml
+        volumeMounts:
+        - name: halyard-config
+          mountPath: /tmp/config
+        - name: halyard-home
+          mountPath: /tmp/spinnaker
+      volumes:
+      {{- if .Values.kubeConfig.enabled }}
+      - name: kube-config
+        secret:
+          secretName: {{ .Values.kubeConfig.secretName }}
+      {{- end }}
+      {{- if .Values.gcs.enabled }}
+      - name: gcs-key
+        secret:
+          secretName: {{ template "spinnaker.fullname" . }}-gcs
+      {{- end }}
+      {{- if and .Values.s3.enabled .Values.s3.accessKey .Values.s3.secretKey }}
+      - name: s3-secrets
+        secret:
+          secretName: {{ template "spinnaker.fullname" . }}-s3
+      {{- end }}
+      - name: halyard-config
+        emptyDir: {}
+      containers:
+      - name: halyard
+        image: {{ .Values.halyard.image.repository }}:{{ .Values.halyard.image.tag }}
+        ports:
+        - containerPort: 8064
+          name: daemon
+        volumeMounts:
+        {{- if .Values.gcs.enabled }}
+        - name: gcs-key
+          mountPath: /opt/gcs
+        {{- end }}
+        {{- if and .Values.s3.enabled .Values.s3.accessKey .Values.s3.secretKey }}
+        - name: s3-secrets
+          mountPath: /opt/s3
+        {{- end }}
+        {{- if .Values.kubeConfig.enabled }}
+        - name: kube-config
+          mountPath: /opt/kube
+        {{- end }}
+        - name: halyard-home
+          mountPath: /home/spinnaker
+        - name: halyard-config
+          mountPath: /opt/halyard/config
+  volumeClaimTemplates:
```

Comment:
Please apply our persistence best practices:
https://github.com/kubernetes/charts/blob/master/REVIEW_GUIDELINES.md#persistence

Here's an example for a statefulset:
https://github.com/kubernetes/charts/blob/master/stable/mongodb-replicaset/templates/mongodb-statefulset.yaml#L313-L339